lib/typeahead/index.js

```diff
@@ -95,15 +100,23 @@ var Typeahead = React.createClass({
 
   getOptionsForValue: function (value, options) {
     if (this._shouldSkipSearch(value)) {
-      return [];
+      return {
+        visible: [],
+        truncated: false
+      };
     }
 
     var searchOptions = this._generateSearchFunction();
     var result = searchOptions(value, options);
+    var resultLength = result.length;
     if (this.props.maxVisible) {
       result = result.slice(0, this.props.maxVisible);
     }
-    return result;
+
+    return {
```

Comment:
I made the change - it was surprisingly small. We also can get away with not calling `slice(0, this.props.maxVisible)` in a couple places, which is nice from an efficiency standpoint. I will end up adding in unit tests where applicable, but the current unit tests are all passing right now.